cmd/nitro/nitro.go

```diff
@@ -309,15 +309,38 @@ func mainImpl() int {
 
 	var l1TransactionOpts *bind.TransactOpts
 	var dataSigner signature.DataSignerFunc
+	var l1TransactionOptsValidator *bind.TransactOpts
+	var l1TransactionOptsBatchPoster *bind.TransactOpts
 	sequencerNeedsKey := nodeConfig.Node.Sequencer.Enable && !nodeConfig.Node.Feed.Output.DisableSigning
 	setupNeedsKey := l1Wallet.OnlyCreateKey || nodeConfig.Node.Staker.OnlyCreateWalletContract
 	validatorCanAct := nodeConfig.Node.Staker.Enable && !strings.EqualFold(nodeConfig.Node.Staker.Strategy, "watchtower")
+	if *l1Wallet != genericconf.WalletConfigDefault && nodeConfig.Node.Staker.L1Wallet != genericconf.WalletConfigDefault && nodeConfig.Node.BatchPoster.L1Wallet != genericconf.WalletConfigDefault {
+		log.Crit("--l1.l1-wallet, --node.staker.l1-wallet and --node.batch-poster.l1-wallet all cannot be set at the same time")
```

Comment:
I'm wondering for cleanliness if we want to just only support `--l1.l1-wallet` if neither of the other two options are set. I think that would make it a little easier to reason about by reducing the number of different valid permutations of options

Reply:
fixed.